reference/normalize.go

```diff
@@ -44,8 +44,13 @@ func ParseNormalizedNamed(s string) (Named, error) {
 	if strings.ToLower(remoteName) != remoteName {
 		return nil, errors.New("invalid reference format: repository name must be lowercase")
 	}
-
-	ref, err := Parse(domain + "/" + remainder)
+	var ref Reference
+	var err error
+	if domain == "" {
```

Comment:
`len(domain) == 0`

Reply:
> len(domain) == 0

Really? I rarely see that as a suggestion for strings?

Reply:
> len(domain) == 0

Taking a quick look through https://golang.org/doc/effective_go.html it seems comparing against `""` is quite idiomatic.

Reply:
Not sure about the docker code but I was scolded few times already for this in origin code base. So I was just passing it on.

Reply:
> Taking a quick look through https://golang.org/doc/effective_go.html it seems comparing against "" is quite idiomatic.

Ok, so using `len` seems to be a kube/origin convention. My apologies.

Looking through this repo I see lots of `== ""` so I propose to leave it as-is.